staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -1268,6 +1302,46 @@ func hasNonAtomicSetType(schema *apiextensions.JSONSchemaProps) bool {
 	})
 }
 
+func requireMapListKeysMapSetValidation(oldCRDSpec *apiextensions.CustomResourceDefinitionSpec) bool {
+	return !hasSchemaWith(oldCRDSpec, hasInvalidMapListKeysMapSet)
+}
+
+func hasInvalidMapListKeysMapSet(schema *apiextensions.JSONSchemaProps) bool {
```

Comment:
I wonder, we could combine the code above and this func, and just check whether the validation error list is empty. I fear a bit these two logics might divert otherwise eventually.

Reply:
You're right. I've combined the logic into a single function.